app/bundles/CampaignBundle/Entity/SummaryRepository.php

```diff
@@ -0,0 +1,162 @@
+            ->groupBy('cs.event_id');
+
+        if ($dateFrom && $dateTo) {
+            $q->andWhere('cs.date_triggered BETWEEN FROM_UNIXTIME(:dateFrom) AND FROM_UNIXTIME(:dateTo)')
+                ->setParameter('dateFrom', $dateFrom->getTimestamp(), \PDO::PARAM_INT)
+                ->setParameter('dateTo', $dateTo->getTimestamp(), \PDO::PARAM_INT);
+        }
+
+        $results = $q->execute()->fetchAll();
+
+        $return = [];
+        // Group by event id
+        foreach ($results as $row) {
+            $return[$row['event_id']] = [
+                0 => intval($row['non_action_path_taken_count']),
+                1 => intval($row['triggered_count']) + intval($row['scheduled_count']),
+            ];
+        }
+
+        return $return;
+    }
+
+    /**
+     * Get the oldest triggered time for back-filling historical data.
+     */
+    public function getOldestTriggeredDate(): ?\DateTime
+    {
+        $qb = $this->getEntityManager()->getConnection()->createQueryBuilder();
+        $qb->select('cs.date_triggered')
+            ->from(MAUTIC_TABLE_PREFIX.'campaign_summary', 'cs')
+            ->orderBy('cs.date_triggered', 'ASC')
+            ->setMaxResults(1);
+
+        $results = $qb->execute()->fetchAll();
+
+        return isset($results[0]['date_triggered']) ? new \DateTime($results[0]['date_triggered']) : null;
+    }
+
+    /**
+     * Regenerate summary entries for a given time frame.
+     *
+     * @throws \Doctrine\DBAL\DBALException
+     */
+    public function summarize(\DateTime $dateFrom, \DateTime $dateTo): void
+    {
+        $sql = 'INSERT INTO '.MAUTIC_TABLE_PREFIX.'campaign_summary '.
+            '(campaign_id, event_id, date_triggered, scheduled_count, non_action_path_taken_count, failed_count, triggered_count) '.
+            '    SELECT '.
+            '        t.campaign_id as campaign_id, '.
+            '        t.event_id as event_id, '.
+            '        FROM_UNIXTIME(UNIX_TIMESTAMP(t.date_triggered) - (UNIX_TIMESTAMP(t.date_triggered) % 3600)) AS date_triggered, '.
+            '        SUM(IF(t.is_scheduled = 1 AND t.trigger_date > NOW(), 1, 0)) as scheduled_count, '.
+            '        SUM(IF(t.is_scheduled = 1 AND t.trigger_date > NOW(), 0, t.non_action_path_taken)) as non_action_path_taken_count, '.
+            '        SUM(IF((t.is_scheduled = 1 AND t.trigger_date > NOW()) OR t.non_action_path_taken, 0, fe.log_id IS NOT NULL)) as failed_count, '.
+            '        SUM(IF((t.is_scheduled = 1 AND t.trigger_date > NOW()) OR t.non_action_path_taken OR fe.log_id IS NOT NULL, 0, 1)) as triggered_count '.
+            '    FROM '.MAUTIC_TABLE_PREFIX.'campaign_lead_event_log t '.
+            '    LEFT JOIN '.MAUTIC_TABLE_PREFIX.'campaign_lead_event_failed_log fe '.
+            '        ON fe.log_id = t.id '.
+            '    WHERE (t.date_triggered BETWEEN FROM_UNIXTIME(:dateFrom) AND FROM_UNIXTIME(:dateTo)) '.
+            '    GROUP BY campaign_id, event_id, date_triggered '.
```

Comment:
@heathdutton I wrote a basic functional test to cover the SummarizeCommand but it does not work as I'd expect. It fails on https://github.com/mautic/mautic/pull/6651/files#diff-c73ead79e78e7652c06c382e160af659c4e6eccca3341bd21834c9c127988768R74

I've added to lead event logs entries for the same campaign, same event, same hour but 2 different contacts. It counts those 2 contacts as 1. It should be 2, right? The summary rows are generated by this query which I haven't touched.